projects/discussion-ui/src/lib/components/lib-entry/lib-entry.component.ts

```diff
@@ -19,30 +22,34 @@ import { ConfigService } from '../../services/config.service';
 export class LibEntryComponent implements OnInit {
 
   data: IdiscussionConfig;
-
+  pageKey: any;
+  config: any;
+  
   constructor(
     public activatedRoute: ActivatedRoute,
     private discussionService: DiscussionService,
-    private configService: ConfigService,
+    private configSvc: ConfigService,
     private location: Location,
+    private navigationServiceService: NavigationServiceService,
     private discussionEventService: DiscussionEventsService,
-    private telemetryUtils: TelemetryUtilsService
+    private telemetryUtils: TelemetryUtilsService,
+    @Inject('configService') protected configService: AbstractConfigService
 
   ) { }
 
   ngOnInit() {
-    this.configService.setConfig(this.activatedRoute);
-    // this.activatedRoute.data.subscribe((data) => {
-    this.data = this.configService.getConfig();
-    if (!this.data) {
-      // fallback for query params
-      this.configService.setConfigFromParams(this.activatedRoute);
-      this.data = this.configService.getConfig();
-    }
-    this.discussionService.userName = _.get(this.data, 'userName');
-    const rawCategories = _.get(this.data, 'categories');
-    this.discussionService.forumIds = _.get(rawCategories, 'result');
-    this.discussionService.initializeUserDetails(this.discussionService.userName);
+    this.navigationServiceService.initService('routerService')
+    this.activatedRoute.queryParams.subscribe((params) => {
+      this.pageKey = _.get(params, 'page')
```

Comment:
for pageKey if u add the interface, we can better understand what data is stored in the pageKey